src/java.net.http/share/classes/jdk/internal/net/http/Http2ClientImpl.java

```diff
@@ -199,10 +199,10 @@ void deleteConnection(Http2Connection c) {
 
     private EOFException STOPPED;
     void stop() {
-        synchronized (this) {stopping = true;}
         if (debug.on()) debug.log("stopping");
         STOPPED = new EOFException("HTTP/2 client stopped");
         STOPPED.setStackTrace(new StackTraceElement[0]);
+        synchronized (this) {stopping = true;}
```

Comment:
Hello Daniel, the other places where these `stopping` and the `STOPPED` get used appears to be when a connection is offered back to the HttpClient pool. So moving this setting of `stopping` appears OK to me.